### ISSUE TITLE
Shrink the binary size

### ISSUE THREAD

**Issue by gliechtenstein** [Jasonette/JASONETTE-Android#13]
The reason it's huge is because of a library that's used to process the JSON template--the end result contains one binary for every possible Android device type.

Find another library that's much smaller and don't have this problem.

**Comment by hakimihamdan88** [comment]
the apk file quite big

**Comment by wallabyway** [comment]
consider using 
https://github.com/borisvanschooten/glesjs/tree/master/libs/armeabi

it's 3MB uncompressed .so, compared to liquidPlayer's 40Mb .so

**Comment by gliechtenstein** [comment]
@wallabyway wow 3MB sounds great! But I just took a look and not sure if this is a full fledged javascript engine or a specialized engine for dealing with WebGL only. According to the repo:

> Gles.js provides Android WebGL bindings for the Javascript V8 engine 

Also,

>  The goal is not to provide a complete Web browser environment, but just a no-frills direct WebGL canvas, similar to what Ejecta does for IOS. 

p.s.
We are definitely moving away from the LiquidCore based engine. In fact I've been experimenting with a couple of other engines last couple of days, and have some promising results. Let me know if I'm missing something about this Gles.js. If it's the right alternative I will definitely try it and consider this.

**Comment by wallabyway** [comment]
The lib file is regular v8 engine. It's specific to arm hence it's small size.  Use it like a regular v8.  The additional webgl bindings are external to the .so file.
If this can run regular three.js, and pixi , i'm pretty sure you can throw anything at it.

**Comment by gliechtenstein** [comment]
Alright guys, starting off 2017 with a slimmer Jasonette!

We switched out the JS engine and now the entire APK size is at 10MB! That's over 75% smaller than before! https://twitter.com/jasonclient/status/815919678409347072

Here's the related PR https://github.com/Jasonette/JASONETTE-Android/pull/42

We took some time to make sure this works for not just the core JSON parser but also other JS files we've been using, including the CSV parser and the RSS parser, thanks to @brad 

There is room for making it even smaller (By splitting binary/giving up on x86) but for now I think 10MB is good enough. If anyone wants to look further into this you're welcome to do so though.
